src/components/common/dialogs/ExplanationDialog.js

```diff
@@ -85,7 +85,13 @@ const ExplanationDialog = ({
       )}
       {(imageSource || ImageComponent) && (
         <View style={[styles.centerImage, imageContainer]}>
-          {ImageComponent ? <ImageComponent {...imageProps} /> : <Image source={imageSource} {...imageProps} />}
+          {ImageComponent ? (
```

Comment:
Some custom (non-svg) components used as Image (e.g. GoodMarket icon) are consuming imageProps.
So this fix will break it's design

The most correct solution is:
1. restore as it was before
2. fix the svg component wrapper:
https://github.com/GoodDollar/GoodDAPP/blob/react-native/config/webpack.config.dev.js#L343
as you could see, here we converting svg file to a component. the template describes the JS source of the component generated 'on the fly'. `props` and `jsx` are just the string. check what's inside, and substitute all props required to the `jsx` using simple string replace. 

For example if `props` will be a string like `{ style, className, some, other, props }` and `jsx` is an formatted svg (XML) code, you could doSomething like:
```javascript
const jsxProps = props.replace(/[\{\}]/, '')
  .split(',')
  .map(prop => proptrim())
  .map(prop => `prop={${prop}}`)

const jsxWithProps = jsx.replace('<svg', `<svg ${svgProps}`)

// ...
return typeScriptTpl.ast`
                    ${imports}
                    ${interfaces}
                    function ${componentName}(${props}) {
                      return ${jsxWithProps};
                    }
                    ${exportLoadedFileAsUrl}
                    export default ${componentName}
                  ` 

```

If props contains just `props` the solution will be much easier:

```javascript
const jsxWithProps = jsx.replace('<svg', `<svg {...${props}}`)

// ...
return typeScriptTpl.ast`
                    ${imports}
                    ${interfaces}
                    function ${componentName}(${props}) {
                      return ${jsxWithProps};
                    }
                    ${exportLoadedFileAsUrl}
                    export default ${componentName}
                  ` 
```

If it requires a lot of time you could try just to pass imageProps also to the ImageComponent as the quick fix - this may solve the issue

Reply:
actually the problem isn't that props aren't passed down to the svg, that's working fine, and because it's passing props down the svg breaks with unrecognised props for svgs. based on your suggestion what I'm trying is to filter props and pass down only the ones that apply to svgs and discard the rest ones

Reply:
@serdiukov-o-nordwhale @mauriguz
lets see if we can change ExplanationDialog to simply receive a component that already defines its own styles and display it as is

Reply:
@sirpy i fixed all custom component displayed as image inside ExplanationDialog and pushed the commit
@mauriguz please check is everything ok with design after that fix